src/cmdlib.sh

```diff
@@ -251,3 +252,65 @@ EOF
     fi
     return "$(cat "${workdir}"/tmp/rc)"
 }
+
+gitart() {
+    # gitart prepares two artifacts from a GIT repo:
+    #   1. JSON describing the GIT tree.
+    #   2. A tarball of the source.
```

Comment:
Is that really required?

Reply:
The docs imply that Koji Content generator should drop enough artifacts to reasonably re-create the build. The docker container effectively is the source recipe, and so I struggled with how much to ship off since the containers that do the build could be deleted.  

To whit: `Along with the content to be managed by Koji, a Content Generator will provide enough metadata to enable a reasonable level of auditing and reproducibility...All inputs to a build task should be preserved either as logs, a database, or as output of the build itself.` [1] 

Obviously, this is a balancing act, since archiving a  700Mb container each time is is a bit heavy handed. The goal of this PR is capture enough information such as the config used for the image and the source code to generate the images, such that a reasonable person would be able to audit and re-produce the build (painful).  

[1] https://docs.pagure.org/koji/content_generators/

Reply:
The container image digest seems to me to be sufficient for this.

I mean, Koji itself isn't today archiving the RPM `BuildRequires:` in some extra place either.

Reply:
The way I'm thinking of this is - we don't rev the production cosa containers too often.  And we try hard to preserve the containers that were deployed to production for quite a while.

Reply:
> The container image digest seems to me to be sufficient for this. I mean, Koji itself isn't today archiving the RPM `BuildRequires:` in some extra place either.

Its worth stating (for the sake of completeness in the conversation), that Content Generation is different than an RPM build. With the RPM build  there is a SRPM from which a person can re-create the RPM and inspect the source.  My thinking with the git archive is that its functionally the SRPM. 

Either way, I'm fine; the value in this PR is that we have a dicussion to reference later.

> The way I'm thinking of this is - we don't rev the production cosa containers too often. And we try hard to preserve the containers that were deployed to production for quite a while.

This, IMHO, is the strongest argument against archiving container source. But what we don't have the `config` source since that's fetched as a separate step. I would argue that at the very least we need to have the COSA container source URL, name and SHA, with a config tarball.

Reply:
So the plan here is to archive the "inputs" to the container build, correct? Which I think you have done. The one thing that is interesting to me is: do we for some reason need to include the list of packages that got installed in the container in the data we archive?